### ISSUE TITLE
Fn::If in AWS::Serverless::Function-Events

### ISSUE THREAD

**Issue by farski** [aws-cloudformation/cfn-lint#2342]
*cfn-lint version:*  0.63.0

*Description of issue.*

I'm seeing a couple weird behaviors with `Fn:If` in the `Events` property of `AWS::Serverless::Function`.

When I do the following, I get a warning that `IsPrimaryRegion` is never used.

```yaml
Conditions:
  IsPrimaryRegion: !Equals [!Ref RegionMode, Primary]

Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri:
        Bucket: !Ref CodeS3Bucket
        Key: !Ref CodeS3ObjectKey
      Events:
        ScheduleTrigger:
          Properties:
            Schedule: !Sub rate(${kScheduleIntervalMinutes} ${kScheduleIntervalLabel})
            Enabled: !If [IsPrimaryRegion, true, false]
          Type: Schedule
```

If I change to putting the entire event in an `If`:

```yaml
Conditions:
  IsPrimaryRegion: !Equals [!Ref RegionMode, Primary]

Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri:
        Bucket: !Ref CodeS3Bucket
        Key: !Ref CodeS3ObjectKey
      Events:
        ScheduleTrigger: !If
          - IsPrimaryRegion
          - Properties:
              Schedule: !Sub rate(${kScheduleIntervalMinutes} ${kScheduleIntervalLabel})
              Enabled: true
            Type: Schedule
          - !Ref AWS::NoValue
```

It complains that `ScheduleTrigger` has no ID. (There are other `Events`, so that property is never empty).

**Comment by chrisoverzero** [comment]
`cfn-lint` is correct here: The `Enabled` property in SAM `Schedule` configuration doesn't work the way it seems like it should. I've [addressed](https://github.com/aws-cloudformation/cfn-lint/issues/2038#issuecomment-867860163) this before, but the short answer is that `Enabled` can't accept an intrinsic – like `!If` – it gets transformed to a constant `ENABLED`.

The SAM team [has added](https://github.com/aws/serverless-application-model/pull/2468) `State`, which is passed through directly in the CloudFormation transformation, but I _think_ it hasn't been released yet. SAM 1.50.0 was released 22 days ago, and that MR was merged _21_ days ago. So close!

**Comment by farski** [comment]
@chrisoverzero Ah, that's an interesting gotcha. Good to know about the forthcoming `State`. Thanks.
